### PR TITLE
with-clerk example updated to app router

### PR DESCRIPTION
**Pull request by devagrawal09**
Updates the `with-clerk` example to use the app router and latest version of `@clerk/nextjs`

### CODE REVIEW

**Comment by devagrawal09** [comment]
@perkinsjr

File: examples/with-clerk/.env.local.sample [hunk 1]
```diff
@@ -0,0 +1,7 @@
+NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=your_clerk_pub_key
+CLERK_SECRET_KEY=your_clerk_secret_key
```
Comment (leerob) [inline examples/with-clerk/.env.local.sample:2]: Why `sample` instead of `example`?

File: examples/with-clerk/app/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode
+}) {
+  return (
+    <html lang="en">
+      <head>
+        <link
```
Comment (leerob) [inline examples/with-clerk/app/layout.tsx:43]: We shouldn't be using `link` tags directly here, it should use the metadata API.

For external CSS files, you should directly import the CSS file in the related component.

File: examples/with-clerk/.env.local.sample (hunk 1 above)
Comment (devagrawal09) [reply examples/with-clerk/.env.local.sample:2]: Mistake, will change this back

File: examples/with-clerk/app/layout.tsx (hunk 2 above)
Comment (devagrawal09) [reply examples/with-clerk/app/layout.tsx]: Fixed, didn't realize any external stylesheets have to be colocated with the code

**Comment by devagrawal09** [comment]
There is now
https://nextjs-clerk-starter-navy.vercel.app/

**Review by leerob — APPROVED** [review]

**Pull request merged**